### ISSUE TITLE
Add coin type name and related account address to events in `coin.move`

### ISSUE THREAD

**Issue by isaacdoidge** [Entropy-Foundation/aptos-core#30]
[The events](https://github.com/Entropy-Foundation/aptos-core/blob/integrate_consensus_key/aptos-move/framework/supra-framework/sources/coin.move#L124) output by the current version of `coin.move` only contain the amount of the coin that was transacted. [Aptos' latest implementation](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L901) adds the type name of the coin implementation and the address of the account performing the action (deposit/withdraw). If we can add this data then clients should be able to track balance changes by monitoring events alone. 

This is advantageous because [the current recommended strategy](https://github.com/Entropy-Foundation/aptos-core/blob/integrate_consensus_key/aptos-move/framework/supra-framework/sources/coin.move#L124) for doing this requires the transaction output results to contain more detailed information about the state changes, which we do not currently store in `smr-moonshot`. It also requires the clients to parse the GUIDs of the events, which are only present in Event-Handle Events. Per a recent discussion, we are planning to adopt Module Events as our standard, so want to avoid having clients depend on deprecated structures. 

To summarize, we should either add the `coin_type` and `account` [fields](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L172) to the existing events in `coin.move`, or move directly to the Module Event model (like the [latest version](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move) of Aptos' contract), which will necessitate adding these fields anyway.

**Comment by sjoshisupra** [comment]
Fixed in #31

**Comment by sjoshisupra** [comment]
Fixed in ab46405dd140b5ebe97cf5afb59299fe9d52c689 
by upgrading to aptos-framework-1.16, it already has `CoinDeposit` and `CoinWithdraw` events with necessary fields and information to allow one to know what kind of coin has been deposited and withdrawn and to/from which address.
